### ISSUE TITLE
Need to intergrate root privileges in menu

### ISSUE THREAD

**Issue by cjg67** [rescatux/rescapp#48]
Hello Adrian,

With rescapp run from the menu, it needs root privileges.  This is not accomplished by solely adding a normal user to the sudoers group, as I have proved this on many different versions of linux over the years. 

This is what I have in part in the sudoers file, obtained by doing visudo:

`##
## Allow root to run any commands anywhere
root    ALL=(ALL)       ALL
cjg     ALL=(ALL)       NOPASSWD: ALL`

Having that works only for issuing commands via the command line.  In a graphical environment there are tools such as kdesudo, gtksudo and ssh-ask password that are required to provide this in graphical environments.  This needs to be added to any livecd/dvd/usb base list of programs.

The same really is true for rescapp.  It also seems that samunlock also requires root rights as well.

Regards,

Christopher.

**Comment by adrian15** [comment]
Can you please tell me which Fedora live cd you use and where to download the rpm you used?

I want to reproduce it. I think this is something about the default Fedora path for root application when queried by non root users.

rescapp is not meant to be run as a root user but as normal user the way I have designed it.

Thank you.

**Comment by cjg67** [comment]
Hello Adrian,

I am certainly interested in hearing about your findings on this, as I have never seen working an application that some part that requires root privs, such as samunlock and selinux as a normal user, launched from a graphical menu, with only adding the user to the sudoers file.

From my own tests, when launched from the menu, there are yellow exclaimation marks next to those two in the self test menu.  When I made the changes then the yellow exclamation marks go away as it is launched with root privileges.

I used the following to install fedora:

https://download.fedoraproject.org/pub/fedora/linux/releases/30/Spins/x86_64/iso/Fedora-LXQt-Live-x86_64-30-1.2.iso

And the following .rpm from the other thread:

https://github.com/rescatux/rescapp/files/3653168/rescapp.zip

Regards,

Christopher.

**Comment by adrian15** [comment]
I just used the live cd, installed myrescue rpm, the python2-sip package (which I think was missing in your package dependencies) and finally your rescapp rpm.

Using rescapp from the menu shows OK for the selinux support. No yellow exclamation mark found.

I did not try to install Fedora as you did though. I just used the live cd as is.

Live cd has:
```
%wheel ALL=(ALL) ALL
```
in its sudoers and liveuser is part of the wheel user group.

The reason why it has no problems is because Fedora setups the liveuser without a password assigned to it.

I have used passwd to change liveuser password. Once I do if I open rescapp from the menu I get the warning signs you describe for selinux.

I use visudo to add:
```
liveuser     ALL=(ALL)       NOPASSWD: ALL
```
and now the warning signs for selinux are gone.

So...

I have already told you how to setup your sudoers properly.
I do not believe you having:
```
cjg     ALL=(ALL)       NOPASSWD: ALL
```
.
Either you are using another user for logging into the desktop, the line has some sort of typo, or you have not logged out and logged in after changing sudoers.

You should be testing is rescapp in the live cd. Not in an installation.

Anyways... if you don't manage to make it work properly in your installation... please insist to me so that I perform the installation.

Last thought: Maybe cjg user is inside wheel group and, somehow you are writing the cjg line before the %wheel line makes %wheel take precedence over the cjg line. In that case try to put the cjg line the last one in the file.

And make sure it does not start with # character. That makes the line a comment.

Hopefully we figure this one out :) .

**Comment by cjg67** [comment]
Hello Adrian,

Yes it is weird.  I am not familiar with the inner workings of a livecd/dvd as I have never created one, and there is definately something different with a livecd and an installed version, as the livecd's environment is read only in so much as when booted from the cd/dvd drive it is not writeable.

I will now make sure that I only test things in the live environment.  It is possible I may have missed something with the fedora 30 installation.  With the vm's that I subsequently installed, in the installation, it had the option to create my cjg user with admin rights, so I used that, and did not manually change anything.

There are differences to how the main stream distributions set things up.

I can also assure you that I have most certainly logged out of the fedora 30 install as I have rebooted my computer a number of times, and as I can use sudo from the command line in which ever terminal that I am using, the sudoers is correctly setup.  The differences were only from a graphical menu.

I have been using linux/unix since 1997 so have used sudo since then.

I am going to test the python3 version of rescapp and create a new rpm, as I have updated the spec file since creating that early version that I uploaded here.

I just need to make sure that the changes to the function that you changed in usr/bin/rescapp has actually been incorporated, and then it will be a fair test.

Regards,

Christopher.

**Comment by adrian15** [comment]
> I can also assure you that I have most certainly logged out of the fedora 30 install as I have rebooted my computer a number of times, and as I can use sudo from the command line in which ever terminal that I am using, the sudoers is correctly setup. The differences were only from a graphical menu.
> 
> I have been using linux/unix since 1997 so have used sudo since then.

Yeah. But do you have the nopasswd line that enables the user to skip typing its password when issuing sudo ? That's the part we need to emulate from a live installation in an installed system.

**Comment by adrian15** [comment]
> I am going to test the python3 version of rescapp and create a new rpm, as I have updated the spec file since creating that early version that I uploaded here.

Nice. Thank you.

Don't miss the latest developments explained here: https://github.com/rescatux/rescapp/issues/46#issuecomment-538783029 .

**Comment by cjg67** [comment]
Hello Adrain,

Well I most certainly did have the `cjg     ALL=(ALL)       NOPASSWD: ALL` when using visudo and it was directly below the root user, however I did not have the %wheel line uncommented.  After uncommenting that, saving and exiting, and then changing the rescapp.desktop file, and removing the sudo -A command, I can confirm that there are no yellow exclamation marks, and everything is useable.

Regards,

Christopher.

**Comment by adrian15** [comment]
Thank you. Now that makes sense. If you are using the default sudoers template you would not need to uncomment the nopasswd wheel line.
Just clone that line, replace %wheel by cjg and uncomment it.
Paste it one line below and you are done.

I guess having your line just after the root line was too above and later %wheel line overwrote it.
Just saying. Anyways now it works as expected and that's the important bit.

No longer need for you to test in live environments. Well, maybe as a last step to confirm your use/installation documentation but, feel free, to continue to develop in an installation setup. Now that the nopasswd line works it has my blessing.

**Comment by cjg67** [comment]
Hello Adrian,

Just created another rpm for the python3 version, and as I am having issues with it, I need to place it here and see if I still get the python errors that I am on my current fedora 30 installation when using it in the livedvd.

Regards,

Christopher.
[rescapp.zip](https://github.com/rescatux/rescapp/files/3695128/rescapp.zip)

**Comment by cjg67** [comment]
Hello Adrian,

Well I have just tested rescapp python3 version via the livedvd and am happy to report that it works perfectly.

I used the following to install the rpm:

sudo dnf install rescapp-0.54b1-1.fc30.x86_64.rpm

using this, I have found that the required dependancies are automaticlly pulled in, with the exeptance of myrescue, because the exernal repository has to be added first.  

This is the first time that I have tested it using dnf, so I am very happy that I have the required files section set out correctly.  I understand that you have made myrescue an optional dependancy, so do you want me to remove it from the spec file?  It really will not hurt leaving it in, but I will do so if you wish.

The above rescapp.zip file will only work with python3 as I stripped the files required for python2.

Regards,

Christopher.

**Comment by adrian15** [comment]
> I understand that you have made myrescue an optional dependancy, so do you want me to remove it from the spec file?

Please remove it. It was never a dependency but it was listed as such by mistake. Please also check the other mentioned dependency changes.

> I have yet to add the chntpw file to the list of required files, as I have yet to make the rpm for it.

My Debian package depends on chntpw package but I don't enforce it to be the rescatux specific one. That way some features might work even if it's not the perfect chntpw.

**Comment by cjg67** [comment]
Hello Adrian,

I think I now have things the way you wanted.  This is the list of files that I have included in the spec file now:

`BuildRequires:make,coreutils        
Requires:python3-qt5,python3-devel,python3-qt5-webkit,python3-sip,dmraid,lvm2,gpart,dosfstools,cryptsetup,cryptsetup-libs,btrfs-progs,reiserfs-utils,xfsprogs,xfsdump,ntfs-3g,gdisk,efibootmgr,python3-pyparted           
Requires:zenity,xdg-utils,hexchat,syslinux,pastebinit,gawk,extundelete,os-prober,mokutil,python3-libselinux,libselinux-utils,policycoreutils-python-utils,policycoreutils-gui,setools-console,selinux-policy-mls,setools,mcstrans`

Regards,

Christopher.

**Comment by adrian15** [comment]
> I think I now have things the way you wanted. This is the list of files that I have included in the spec file now:

I think you are just missing the chntpw dependency.

**Comment by cjg67** [comment]
Hello Adrian,

I am in the process of creating a page describing how to install rescapp on fedora.  I have not yet completed it, but you can view the progress here:

https://github.com/cjg67/rescapp/wiki/Installing-Rescapp-on-Fedora

Once it is complete, I can remove certain files from the spec file, as people will then be able to choose which of the optional files that they need for their particular circumstances, such as gdisk myrescue etc.  I know you said that you would host the page, but I needed someware that I could work on it, that would be compatible with what you already have, so the easiest solution was to use the wiki provided on github.

I know the chntpw dependency is not included yet.  I am going to test chntpw that is provided with fedora on the livedvd, as even with it installed on my current installation of fedora 30 it is not recognised by rescapp self-test, and as I had made many changes to the installed os, it is not a fair test.  I will also create the rpm from your patched source.

Regards,

Christopher.

**Comment by cjg67** [comment]
Hello Adrian,

I have created an account on sourceforge and have copied the installation instructions to the wiki there as well.  I can upload the rpm's there, once you are happy that everything has been included.  

I have verified that the chntpw package that is included in the fedora repositories does not appear to include samunlock for some reason or other.  It shows up even on the livedvd as not found in the rescapp self-test.

I will create a folder for rescapp rpm and for chntpw rpm on my sourceforge account.

You can view the progress on the sourceforge page here:

[https://sourceforge.net/p/rescapp-fedora/wiki/Install%20Rescapp%20on%20Fedora/](url)

Regards,

Christopher.

**Comment by adrian15** [comment]
> I am in the process of creating a page describing how to install rescapp on fedora. I have not yet completed it, but you can view the progress here:
> 
> https://github.com/cjg67/rescapp/wiki/Installing-Rescapp-on-Fedora

1) Why do you install manually the normal dependencies?

I use yum (I think the same can be done with dmf) like this:

```
yum install ./rescapp-something.rpm
```

and it pulls all of the dependencies you setup in the rpm.

No need to setup a repo and no need to install dependencies manually.

2) The part where you explain how to install optional dependencies it's fine to me.

3) Finally if you don't modify rescapp source code, which I think you don't then this is not a rescapp fork but just you packaging rescapp for Fedora.

**Comment by adrian15** [comment]
> Once it is complete, I can remove certain files from the spec file, as people will then be able to choose which of the optional files that they need for their particular circumstances, such as gdisk myrescue etc.

Well, you should not messing around with dependencies so much.
Just try to mimic the current ones found on the debian-dev branch.

Then, finally, just add instructions on how to install the optional dependencies (the ones you already have for gparted, gpart and testdisk).

**Comment by adrian15** [comment]
> I am in the process of creating a page describing how to install rescapp on fedora. I have not yet completed it, but you can view the progress here:
> 
> https://github.com/cjg67/rescapp/wiki/Installing-Rescapp-on-Fedora
> 
> SeLinux Support

I have just found out the SeLinux Support section. Somehow I believed SeLinux packages were part of the normal rescapp dependencies. They are not. And that was made on purpose because not every SeLinux live cd has Selinux support.

Fedora has it by default because it boots in SeLinux mode but it might not have all the SeLinux tools so that Rescapp can fix SeLinux systems. And that's what you are bringing with the installation command.

That's fine.

Now I'm starting to think about needing two pages.

One for the final user as the one I did (with a big one liner). And another one for the live cd integrator / builder which wants to use your rescapp and chntpw packages in its live cd.

The live cd builder instructions page can wait for now.

**Comment by adrian15** [comment]
> I know the chntpw dependency is not included yet. I am going to test chntpw that is provided with fedora on the livedvd, as even with it installed on my current installation of fedora 30 it is not recognised by rescapp self-test, and as I had made many changes to the installed os, it is not a fair test. I will also create the rpm from your patched source.

Yeah, in the end you need to package my patched chntpw, install it and then install your rescapp package. As your packaged chntpw might have an slightly superior version to the Fedora one when installing rescapp you'll keep your installed patched chntpw package in place.

**Comment by adrian15** [comment]
> I have created an account on sourceforge and have copied the installation instructions to the wiki there as well. I can upload the rpm's there, once you are happy that everything has been included.

I think I already told you this. You don't need sourceforge for hosting these files. I think you can github binary releases for them. Anyways just use whatever seems better for you. Although I will host your instructions your rpms need to be hosted somewhere else (so that they are not official ones and I don't have to maintain them myself).

**Comment by adrian15** [comment]
> I have verified that the chntpw package that is included in the fedora repositories does not appear to include samunlock for some reason or other.

Yeah, the samunlock binary does not exist in upstream chntpw. It's an addition of my patch/fork so that accounts can be unlocked in an automatic way.

As Debian does not want my patches I might end making a proper fork so that my fixed chntpw can be included into Debian (and other distros) without too much hassle.

I will focus on making both rescapp and rescatux stable for now tough.

**Comment by cjg67** [comment]
Hello Adrian,

Fedora are in the process of removing yum. dnfdragora is their official application now days.  When I first started this project, I was using yum.

The fact is, I would not have added the instructions for any of these packages, as they can all be left within the spec file, and it would install all of them.  No one will ever see what is in the spec file.  The only package that needs to have instructions for it is myrescue.  It was only because you said to look at the changes to the packages that you made, that I started down that path.

I hate the fact that github will not allow .rpm files unless they are zipped.  If I use tito, which is actually the correct way when building rpm's from git,  then I need to know that the binary and source rpm's are not going to be rejected by their system.

I just do not know what direction to go in now.  I would prefer to only have the instructions for myrescue, and have everything else, regardless of if it is a system addition or optional, within the spec file and it gets installed.  

Please advise exactly what you wish me to do.  I do not mind either way.  I am a darn good technical writer as I have written server setup instructions for webhosting companies and also for a technical magazine.

Regards,

Christopher.

**Comment by adrian15** [comment]
> Fedora are in the process of removing yum. dnfdragora is their official application now days. When I first started this project, I was using yum.

Just use dnf then. I think it works the same way. Installing a local rpm should pull out  the dependencies from the repo.

> The fact is, I would not have added the instructions for any of these packages, as they can all be left within the spec file, and it would install all of them. No one will ever see what is in the spec file.

Yeah, having those instructions is fine for your main page in order to describe your rescapp-fedora project.
However I will probably not add those normal dependencies installation instructions into my own documentation on how to use your rpms. I will use the dnf command that install local rpm and pulls other dependencies from the repo.

> I hate the fact that github will not allow .rpm files unless they are zipped.

That's why you need sourceforge. Ok.
Did you try github binary releases (which it's a bit different than normal releases) ? Not sure it's going to work but you can try there. Or maybe they have something about hosting a webpage in github and you can put the rpms there.

Just advising... I mean... you can put rpms wherever you want. I just wanted you to only have one account so that it's more simple for you.

> I would prefer to only have the instructions for myrescue

myrescue is not a rescapp dependency.
It's not need at all by rescapp.
Rescapp does not try to run it.
Just remove it alongside gddrescue and forget about it.

**Comment by adrian15** [comment]
> Please advise exactly what you wish me to do.

- Main page about your project is about (Links to the other two pages)
- Instructions for Debian live integrators/builders about how it's the best to integrate rescapp into their live cds (making sure they use the patched chntpw version and also making sure all the selinux dependencies are there).
- Instructions for non-free Rescatux users

For these last instructions which I would almost copy-and-paste into official Rescatux documentation you should have:

- How to get the Fedora live cd

And one-liner command for doing all of the following in one go:

- How to install SeLinux packages
- How to download and install custom chntpw package
- How to download and install rescapp package
.

**Comment by adrian15** [comment]
> and have everything else, regardless of if it is a system addition or optional, within the spec file and it gets installed.

No. That's not right. The spec file should only have whatever there is in Debian counterpart. That's because I do not want to enforce neither rescatux's chntpw nor selinux support.

If any of those two are missing self-test is going to warn you but you can always use the other rescapp options without being constrained by the dependencies. That way, even if integration is not perfect, it makes easier for rescapp to be included in other live cds.

I understand why you would like everything in the spec file. It would make documentation much more simple to write.

**Comment by cjg67** [comment]
Hello Adrian,

I am now getting a clearer picture of what you would like.  I only created those pages so that you could cherry pick what you want.  

My method of technical writting, that I did when working for a webhosting company, was to write the manual for server set ups in a way that the least technical person in the company could follow, if he wished to do a server install.  That person happened to be the owner.  I then gave him the manual to read, and he loved it.  It took me 72 hours to go through a complete server installation, taking notes and then re-writeing it and then using the finished documentation to setup a new server from scratch.

I have the time to assist you with cleaning up/adding to your existing documentation.

I am unable to test your instructions for using chntpw on a windows installation as I do not have a computer or server with windows installed on it.

I will make this committment as well,  I  will continue to package the rpms for as long as you remain interested/have time for working on rescapp.  Once the instructions have been completed, should you find that you pass the project on to another developer, then that would be up to them to continue on with the rpm's or not.  The spec files for the packages would remain, and they would need to do it themselves.

Regards,

Christopher.

**Comment by adrian15** [comment]
> I am now getting a clearer picture of what you would like. I only created those pages so that you could cherry pick what you want.

Ok.

> My method of technical writting, that I did when working for a webhosting company, was to write the manual for server set ups in a way that the least technical person in the company could follow, if he wished to do a server install. That person happened to be the owner. I then gave him the manual to read, and he loved it. It took me 72 hours to go through a complete server installation, taking notes and then re-writeing it and then using the finished documentation to setup a new server from scratch.

Whenever I can I use the same method. Now it's easier to start from a new virtual machine or virtual machine snapshot to test if your documentation does not miss any step.

> I have the time to assist you with cleaning up/adding to your existing documentation.

That would be a great thing. I need to write a lot of the embedded documentation for rescapp. Once that's done the documentation being reviewed by a native English speaker should improve it a lot.

> I am unable to test your instructions for using chntpw on a
> windows installation as I do not have a computer or server with windows installed on it.

You can always install Windows 10 in a virtual machine. Nowadays it works even if you don't activate it. At least it should be useful for the tests.

Anyways I can always test your chntpw on my test virtual machine. Even if I don't build the rpms myself it doesn't mean that I cannot test them.

> I will make this committment as well, I will continue to package the rpms for as long as you remain interested/have time for working on rescapp.

That's very nice from you. My idea is to ping you only when a new important rescapp release is out.

> Once the instructions have been completed, should you find that you pass the project on to another developer, then that would be up to them to continue on with the rpm's or not. The spec files for the packages would remain, and they would need to do it themselves.

Yeah, the most important software part is documentation. So that developers can continue to improve the software.
